File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dictionary = {}
        value_array = []
        result =[]
        for num in nums:
            if num in dictionary:
                dictionary[num]+=1
            else:
                dictionary[num] =1
        for value in dictionary.values():
            value_array.append(value)
        # value_array.sort(value_array.begin(),value_array.end())
        value_array.sort()
        k_freq = value_array[len(value_array)-k]
        for key,value in dictionary.items():
            if value>=k_freq:
                result.append(key)    
            
        return result

File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length()!=t.length()) return false;
        int sfreq[26] ={0};
        int tfreq[26] ={0};
        for(int i=0;i<s.length();i++){
            sfreq[s[i]%97] +=1;
            tfreq[t[i]%97] +=1;
        }
        for(int i=0;i<26;i++){
            if(sfreq[i]!=tfreq[i]){
                return false;
            }
        }
        // sort(s.begin(),s.end());
        // sort(t.begin(),t.end());
        // int n = s.length();
        // for(int i=0;i<n;i++){
        //     if(s[i]!=t[i]){
        //         return false;
        //     }
        // }
        return true;
        
    }
};
